tests/check_check_sub.c

```diff
@@ -497,6 +497,511 @@ START_TEST(test_ck_assert_uint_expr)
+  x*=10.0;
+  y*=10.0;
+  ck_assert_floating_ge_tol10(y, x, -2);
+  ck_assert_floating_ge_tol10(x, x, -2);
+  ck_assert_floating_ge_tol10(y, y, -2);
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ge_tol10(x, y, -2);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_ge_tol10_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  int p = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ge_tol10(2%d, 3%f, -1%p);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_le_tol10)
+{
+  record_test_name(tcase_name());
+
+  double x = 0.001;
+  double y = 0.003;
+  ck_assert_floating_le_tol10(x, y, -3);
+  ck_assert_floating_le_tol10(x, x, -3);
+  ck_assert_floating_le_tol10(y, y, -3);
+  x*=10.0;
+  y*=10.0;
+  ck_assert_floating_le_tol10(x, y, -2);
+  ck_assert_floating_le_tol10(x, x, -2);
+  ck_assert_floating_le_tol10(y, y, -2);
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_le_tol10(y, x, -2);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_le_tol10_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  int p = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_le_tol10(3%d, 2%f, -1%p);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_finite)
+{
+  record_test_name(tcase_name());
+
+  double x = 0.0001;
+  ck_assert_floating_finite(x);
+  // MS VS doesn't allow explicit division by zero
+  double t = 1;
+  x = 1.0 / (1.0 - t);
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_finite(x);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_finite_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  double t = 1;
+  double x = 1.0 / (1.0 - t);
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_finite(x*(1%d));
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_infinite)
+{
+  record_test_name(tcase_name());
+
+  double t = 1;
+  double x = 1.0 / (1.0 - t);
+  ck_assert_floating_infinite(x);
+  x = -1.0 / (1.0 - t);
+  ck_assert_floating_infinite(x);
+  x = 0;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_infinite(x);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_infinite_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_infinite(2%d);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_nan)
+{
+  record_test_name(tcase_name());
+
+  double t = 1;
+  double x = 0.0 / (1.0 - t);
+  ck_assert_floating_nan(x);
+  x = 1.0 / (1.0 - t);
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_nan(x);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_nan_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_nan(2%d);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_nonnan)
+{
+  record_test_name(tcase_name());
+
+  double t = 1;
+  double x = 0;
+  ck_assert_floating_nonnan(x);
+  // TODO: test with 0.0/0.0 and platform specific output.
```

Comment:
What could be done so that these could be tested? You mentioned that NaN could be negative or positive. Is it possible to ensure that if the value is NaN that it is positive before printing it out? Or, is it simply that some platforms will always print NaN is negative?